src/components/tasks/card/progressBar/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { ProgressContainerProps } from '@/interfaces/task.type';
+
+const ProgressContainer: FC<ProgressContainerProps> = ({ content }) => {
+    const router = useRouter();
+    const { dev } = router.query;
+
+    const [progress, setProgress] = useState<boolean>(false);
+    const [progressValue, setProgressValue] = useState<number>(0);
+
+    const { isUserAuthorized } = useUserData();
+    const { data } = useGetUserQuery();
+    const [updateTask, { isLoading: isLoadingUpdateTaskDetails }] =
+        useUpdateTaskMutation();
+    const [updateSelfTask, { isLoading: isLoadingSelfTaskUpdate }] =
+        useUpdateSelfTaskMutation();
+
+    const { SUCCESS, ERROR } = ToastTypes;
+
+    const handleSliderChangeComplete = async (
+        id: string,
+        percentCompleted: number
+    ) => {
+        const data = {
+            percentCompleted: percentCompleted,
+        };
+        if (isUserAuthorized) {
+            await updateTask({
+                task: data,
+                id: id,
+            })
+                .unwrap()
+                .then(() => toast(SUCCESS, PROGRESS_SUCCESSFUL))
+                .catch(() => toast(ERROR, ERROR_MESSAGE));
+        } else {
+            await updateSelfTask({ task: data, id: id })
+                .unwrap()
+                .then(() => toast(SUCCESS, PROGRESS_SUCCESSFUL))
+                .catch(() => toast(ERROR, ERROR_MESSAGE));
+        }
+    };
+
+    const handleDebounceProgress = () => {
+        handleSliderChangeComplete(content.id, progressValue);
+        setProgress(false);
+    };
+
+    const debounceSlider = (debounceTimeOut: number) => {
+        if (debounceTimeOut) {
+            clearTimeout(debounceTimeOut);
+        }
+        setTimeout(() => {
+            handleDebounceProgress();
+        }, 1000);
+    };
+
+    const handleProgressChange = (
+        event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+        setProgressValue(Number(event.target.value));
+    };
+
+    const handleProgressUpdate = () => {
+        if (
+            content.assignee === data?.username ||
+            data?.roles.super_user === true
+        ) {
+            setProgress(true);
+        } else {
+            toast(ERROR, 'You cannot update progress');
+        }
+    };
+    return (
+        <>
+            <div className={classNames.progressContainerUpdated}>
+                <HandleProgressbar
+                    progress={progress}
+                    progressValue={progressValue}
+                    percentCompleted={content.percentCompleted}
+                    handleProgressChange={handleProgressChange}
+                    debounceSlider={debounceSlider}
+                    startedOn={content.startedOn}
+                    endsOn={String(content.endsOn)}
+                    isLoading={
+                        isLoadingUpdateTaskDetails || isLoadingSelfTaskUpdate
+                    }
+                />
+            </div>
+            {dev === 'true' && (
```

Comment:
Is this a string? Shouldn't it be a boolean variable? I understand it might be something existing that you are re-using here. 

I would suggest to create a getter function in some common util file called `isDevMode` or something which internally does exactly this and can be used across the repo (can be adopted across repo later on, maybe add a ticket for it?).

Reply:
ticket is already created for it
https://github.com/Real-Dev-Squad/website-status/issues/712